Add #torrent-set-location support

qq99 opened the pull request:
- adds ability to set the location of the torrent
- adds ability to `find([id1, id2])` with multiple IDs

qq99 commented:
@fguillen

fguillen commented on lib/transmission_api/client.rb:
```diff
@@ -39,21 +39,41 @@ def all
 
-  def find(id)
+  def find(ids)
     log "get_torrent: #{id}"
```
`id` now is `nil`

fguillen commented on lib/transmission_api/client.rb:
```diff
@@ -39,21 +39,41 @@ def all
         }
       )
 
     response["arguments"]["torrents"].first
```
So you are _finding_ a bunch of torrents but you only return reference to one of them

fguillen commented on lib/transmission_api/client.rb:
```diff
@@ -39,21 +39,41 @@ def all
+
+    response["arguments"]
+  end
+
```
I don't see _test_ for this new method

fguillen commented:
Hi @qq99, I can't accept this PR. The new `find` method is not coherent with the return value if it receives a bunch of ids. And if you make it coherent and make it to return a bunch of references it will break actual integrations. If you really want this functionality make a new method called `find_all` or something like this.

There are no tests for the new functionalities.

Check the comments.

qq99 commented:
Gotcha, will fix :+1:

qq99 commented:
@fguillen, how do you run the tests?
I'm a bit new to gem dev

qq99 commented:
```
ruby -Itest client_test.rb 
Warning: you should require 'minitest/autorun' instead.
Warning: or add 'gem "minitest"' before 'require "minitest/autorun"'
From:
  /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:1:in `<top (required)>'
  /home/anthony/github/TransmissionApi/test/test_helper.rb:1:in `<top (required)>'
  client_test.rb:1:in `require_relative'
  client_test.rb:1:in `<main>'
MiniTest::Unit::TestCase is now Minitest::Test. From /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit/testcase.rb:8:in `<module:Unit>'
/home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:676:in `<class:Runner>': undefined method `_run_suite' for class `Test::Unit::Runner' (NameError)
    from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:261:in `<module:Unit>'
    from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:15:in `<module:Test>'
    from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:7:in `<top (required)>'
    from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
    from /home/anthony/github/TransmissionApi/test/test_helper.rb:1:in `<top (required)>'
    from client_test.rb:1:in `require_relative'
    from client_test.rb:1:in `<main>'
```

fguillen commented:
Yep, test were broken in the version you have forked. Now they are fixed.

You have to merge my master with yours. A possible chain of commands can be:

```
git remote add fguillen git@github.com:fguillen/TransmissionApi.git
git fetch fguillen
git merge fguillen/master master
# Solve conflicts
git add .
git commit -m "Merging master"
# Install gems
bundle
# Run tests
rake
```

On Wed, Aug 27, 2014 at 2:19 AM, Anthony Cameron notifications@github.com
wrote:

> ruby -Itest client_test.rb
> Warning: you should require 'minitest/autorun' instead.
> Warning: or add 'gem "minitest"' before 'require "minitest/autorun"'
> From:
>   /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:1:in `<top (required)>'
>   /home/anthony/github/TransmissionApi/test/test_helper.rb:1:in`<top (required)>'
>   client_test.rb:1:in `require_relative'
>   client_test.rb:1:in`<main>'
> MiniTest::Unit::TestCase is now Minitest::Test. From /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit/testcase.rb:8:in `<module:Unit>'
> /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:676:in`class:Runner': undefined method `_run_suite' for class`Test::Unit::Runner' (NameError)
>     from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:261:in `<module:Unit>'
>     from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:15:in`module:Test'
>     from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/test/unit.rb:7:in `<top (required)>'
>     from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in`require'
>     from /home/anthony/.rbenv/versions/2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
>     from /home/anthony/github/TransmissionApi/test/test_helper.rb:1:in`<top (required)>'
>     from client_test.rb:1:in `require_relative'
>     from client_test.rb:1:in`<main>'
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/fguillen/TransmissionApi/pull/2#issuecomment-53511613
> .

## 

Fernando Guillén
Freelance Web Developer
http://www.fernandoguillen.info

qq99 commented:
Awesome, thanks!  I'll hope to have this done soon

qq99 commented:
Rebased and squashed off your latest master
Still not able to run the tests :(  I'm using `bundle exec ruby -Itest test/client_test.rb`

```
TransmissionApi/test/test_helper.rb:2:in `require': cannot load such file -- mocha/mini_test (LoadError)
    from /home/anthony/github/TransmissionApi/test/test_helper.rb:2:in `<top (required)>'
    from test/client_test.rb:1:in `require_relative'
    from test/client_test.rb:1:in `<main>'
```

this is after a fresh clone & bundle install

fguillen commented:
What your `bundle exec gem list` says?

Looks like you have not the `mocha` gem installed.

My one says:

```
*** LOCAL GEMS ***

bundler (1.6.3)
httparty (0.9.0)
metaclass (0.0.4)
mocha (0.13.0)
multi_json (1.10.1)
multi_xml (0.5.5)
transmission_api (0.0.7)
```

qq99 commented:
```
*** LOCAL GEMS ***

bundler (1.7.2)
httparty (0.9.0)
metaclass (0.0.4)
mocha (0.13.0)
multi_json (1.10.1)
multi_xml (0.5.5)
transmission_api (0.0.7)
```

`ruby 2.0.0p247 (2013-06-27 revision 41674) [x86_64-linux]`

fguillen commented:
It was a bug.. the `minitest` and `mocha` versions were too old.. I have updated them.. please _remerge_ my fguillen/master branch and try again.

Remember to `bundle` first to update the gems.

Also you can use `bundle exec rake` to run all the tests.

qq99 commented:
Thanks, I'll let you know how that goes later tonight
